Add TL files

rlwcomposto opened the pull request:
These files can be used to find lakes in the tangible landscape (TL) using the elevation in the tangible landscape. When the TL is manipulated, these files can be run and the updated lakes will be shown in the TL. The function came from the GRASS GIS wikipage - https://grasswiki.osgeo.org/wiki/Using_GRASS_GIS_through_Python_and_tangible_interfaces_(workshop_at_FOSS4G_NA_2016)#Tangible_Landscape.

rlwcomposto committed:
Add TL files

These files can be used to find lakes in the tangible landscape (TL) using the elevation in the tangible landscape. When the TL is manipulated, these files can be run and the updated lakes will be shown in the TL. The function came from the GRASS GIS wikipage - https://grasswiki.osgeo.org/wiki/Using_GRASS_GIS_through_Python_and_tangible_interfaces_(workshop_at_FOSS4G_NA_2016)#Tangible_Landscape.

rlwcomposto committed:
Updated files

Made new files that addressed errors in the tests.

rlwcomposto committed:
Update to JSON file due to errors

Tried to address errors for JSON file

rlwcomposto committed:
Change JSON to right output

Change JSON file to have the map output be correct ("map=output_lake")

rlwcomposto commented on activities/RC_TL_v2.json:
```diff
@@ -0,0 +1,20 @@
+  "tasks": [
+    {
+      "layers": [
+        ["d.rast", "map=slope"]
```
Thank you for catching that, made the change!

petrasovaa commented:
Congrats, the checks succeeded! Now try to use your geospatial creativity to customize the copied example. For example:
* display the source point (make it into vector first)
* set the water level dynamically as 50th or 75th percentile of the elevation range
* dynamically find the lowest spot and spread water from there
* something else

petrasovaa reviewed (changes requested):
See my last comment.

rlwcomposto committed:
Added some colors to the output

Added streams (blue), added slope (color ramp) and kept lakes. Together can see streams and lakes and how they relate to slope.

rlwcomposto committed:
Update to json

Update to json file to address "File content / JSON files (pull_request) Failing after 5s" error

rlwcomposto committed:
Update to json and python

"env" not recognized in "r.colors" so moved it

rlwcomposto committed:
Update python file

Deleted error in r.colors

rlwcomposto committed:
Changes to json

Update json to display multiple layers

rlwcomposto committed:
Simplifying code

Simplified code to be just extracting streams and lakes

rlwcomposto committed:
Updated error

Added running the function to add the streams to the json file

rlwcomposto committed:
Remove ,

Removing trailing commas

rlwcomposto committed:
Commas were maybe necessary

Added them back in

rlwcomposto committed:
Removed commas again

Removed commas again because that made removed an error

rlwcomposto commented:
Hi Anna, sorry for the delay, I didn't realize a change was requested and what that entailed. I just added a little flair to the code  (some colors and rivers). 

I'm running into some consistent errors in the tests that I haven't been able to fix yet.

petrasovaa commented:
> Hi Anna, sorry for the delay, I didn't realize a change was requested and what that entailed. I just added a little flair to the code (some colors and rivers).
> 
No worries, I think we could do a quick Tangible Landscape session next week.

> I'm running into some consistent errors in the tests that I haven't been able to fix yet.

The relevant error shows incorrect parameter names:

```
ERROR: r.stream.extract: Sorry, <output> is not a valid parameter
ERROR: r.stream.extract: Sorry, <coordinates> is not a valid parameter
ERROR: Required parameter <threshold> not set:
	(Minimum flow accumulation for streams)
```

rlwcomposto committed:
Update r.stream

Add threshold

rlwcomposto committed:
Update .py

Delete output and coordinates

rlwcomposto committed:
Use black

Use black on code

rlwcomposto committed:
Change threshold

rlwcomposto committed:
Update RC_TL_v2.json

Update json to be different from template

rlwcomposto committed:
Change color

rlwcomposto committed:
Changes to json

format to look like earlier one without errors

rlwcomposto commented:
Changes are complete, the map should display lakes and rivers at the same time in blue.

rlwcomposto commented:
Mistakenly closed pull request, opening so the code can be reviewed.
